pkg/cluster/flux_test.go

```diff
@@ -0,0 +1,176 @@
+package cluster_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	"github.com/aws/eks-anywhere/pkg/cluster"
+)
+
+const (
+	EksaGitPrivateKeyTokenEnv = "EKSA_GIT_PRIVATE_KEY"
+	EksaGitKnownHostsFileEnv  = "EKSA_GIT_KNOWN_HOSTS"
+	RsaAlgorithm              = "rsa"
+	EcdsaAlgorithm            = "ecdsa"
+	Ed25519Algorithm          = "ed25519"
+)
+
+type testContext struct {
+	privateKeyFile         string
+	isPrivateKeyFileSet    bool
+	gitKnownHostsFile      string
+	isGitKnownHostsFileSet bool
+}
+
+func (tctx *testContext) SaveContext() {
+	tctx.privateKeyFile, tctx.isPrivateKeyFileSet = os.LookupEnv(EksaGitPrivateKeyTokenEnv)
+	tctx.gitKnownHostsFile, tctx.isGitKnownHostsFileSet = os.LookupEnv(EksaGitKnownHostsFileEnv)
+	os.Setenv(EksaGitPrivateKeyTokenEnv, "my/private/key")
```

Comment:
Curious, why do we need to set these env vars? All the validations in the package `cluster` are supposed to be context agnostic and reading env vars is definitely context dependent.